sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -207,4 +224,55 @@ export class ConfirmationOfEnrollmentController {
     // Send a message to allow the workflow to proceed.
     await this.workflow.sendConfirmCOEMessage(application.assessmentWorkflowId);
   }
+
+  /**
+   * Deny the Confirmation Of Enrollment(COE), defining the
+   * COE status as Declined in the student application table.
+   * @param locationId location that is completing the COE.
+   * @param applicationId application id to be updated.
+   * @param payload contains the denied reason of the
+   * student application.
+   */
+  @HasLocationAccess("locationId")
+  @Patch(
+    ":locationId/confirmation-of-enrollment/application/:applicationId/deny",
+  )
+  async denyConfirmationOfEnrollment(
+    @Param("locationId") locationId: number,
+    @Param("applicationId") applicationId: number,
+    @Body() payload: DenyConfirmationOfEnrollmentDto,
+  ): Promise<void> {
+    try {
+      await this.applicationService.setDeniedReasonForConfirmationOfEnrollment(
+        applicationId,
+        locationId,
+        payload.coeDenyReasonId,
+        payload.otherReasonDesc,
+      );
+    } catch (error) {
+      if (error.name === COE_REQUEST_NOT_FOUND_ERROR) {
+        throw new UnprocessableEntityException(error.message);
+      }
+
+      throw new InternalServerErrorException(
+        "Error while denying a  Confirmation Of Enrollment (COE).",
+      );
+    }
+  }
+
+  /**
+   * Get all COE denied reason ,which are active
+   * @returns COE denied reason list
+   */
+  @Get("confirmation-of-enrollment/denial-reasons")
+  async getCOEDeniedReason(): Promise<COEDeniedReasonDto[]> {
```

Comment:
Just a suggestion, we already have on API and Vue a key/value pair object name OptionItemDto to map this type of return. If this is something that could be changing in the future we can keep creating specific classes. I would be ok either way.

Reply:
I recommend the same as we are following the same standard for almost other dropdowns as well.

Reply:
the radio button option accepts a list of objects with keys, `value` , `label` and `shortcuts`

Reply:
The API can still return the OptionItemDto and the map could take care of providing to the radio button the object that it expects. We are not going to return the API DTO specific to populate a radio or dropdown. The API could return a specific key/value object and the adaptions could be done on Vue application.
Either way, it was a suggestion 😉

Reply:
to avoid extra code mapping, in the `web` , I returned the expected list of objects from the API :)

Reply:
updated as disussed